plugin/trino-mongodb/src/test/java/io/trino/plugin/mongodb/TestMongoConnectorTest.java

```diff
@@ -122,6 +119,29 @@ protected TestTable createTableWithDefaultColumns()
         return abort("MongoDB connector does not support column default values");
     }
 
+    @Test
+    public void testMergeWithCustomizeTypeIdColumn()
+    {
+        String targetTable = "merge_with_customize_type_id_column_target_" + randomNameSuffix();
+        String sourceTable = "merget_with_customize_type_id_column_source_" + randomNameSuffix();
+        createTableForWrites("CREATE TABLE %s (_id INT, customer VARCHAR, purchases INT, address VARCHAR)", targetTable, Optional.empty());
+
+        assertUpdate("INSERT INTO %s (_id, customer, purchases, address) VALUES (1, 'Aaron', 5, 'Antioch'), (2, 'Bill', 7, 'Buena'), (3, 'Carol', 3, 'Cambridge'), (4, 'Dave', 11, 'Devon')".formatted(targetTable), 4);
+
+        createTableForWrites("CREATE TABLE %s (_id INT, customer VARCHAR, purchases INT, address VARCHAR)", sourceTable, Optional.empty());
+
+        assertUpdate("INSERT INTO %s (_id, customer, purchases, address) VALUES (1, 'Aaron', 6, 'Arches'), (3, 'Ed', 7, 'Etherville'), (4, 'Carol', 9, 'Centreville'), (7, 'Dave', 11, 'Darbyshire')".formatted(sourceTable), 4);
+
+        assertUpdate("MERGE INTO %s t USING %s s ON (t._id = s._id)".formatted(targetTable, sourceTable) +
+                "    WHEN MATCHED AND s.address = 'Centreville' THEN DELETE" +
+                "    WHEN MATCHED THEN UPDATE SET purchases = s.purchases + t.purchases, address = s.address" +
+                "    WHEN NOT MATCHED THEN INSERT (_id, customer, purchases, address) VALUES(s._id, s.customer, s.purchases, s.address)", 4);
+        assertQuery("SELECT _id, customer, purchases, address FROM " + targetTable, "VALUES (1, 'Aaron', 11, 'Arches'),  (2, 'Bill', 7, 'Buena'),  (3, 'Carol', 10, 'Etherville'), (7, 'Dave', 11, 'Darbyshire')");
+
+        assertUpdate("DROP TABLE " + sourceTable);
+        assertUpdate("DROP TABLE " + targetTable);
+    }
+
```

Comment:
Could you add a test to update _id column? I think the current implementation is broken:
```sql
CREATE TABLE test (_id int, data int);
INSERT INTO test VALUES (1,10), (2,20);
UPDATE test SET _id = 999 WHERE _id = 1;
SELECT * FROM test;

 _id | data
-----+------
   1 |   10
   2 |   20
(2 rows)
```